File: solutions/javascript/Triangle/Triangle.js
/**
 * @param {number[][]} triangle
 * @return {number}
 */
var minimumTotal = function(triangle) {
    let n = triangle.length, arr = triangle[n - 1];
    for (let i = n - 2; i >= 0; i--) {
        for (let j = 0; j <= i; j++) {
            arr[j] = Math.min(arr[j], arr[j + 1]) + triangle[i][j];
        }
    }
    return arr[0];
};

File: solutions/java/Triangle/Triangle.java
class Solution {
    public int minimumTotal(List<List<Integer>> triangle) {
        int n = triangle.size();
        List<Integer> list = triangle.get(n - 1);
        for (int i = n - 2; i >= 0; i--) {
            for (int j = 0; j <= i; j++) {
                list.set(j, Math.min(list.get(j), list.get(j + 1)) + triangle.get(i).get(j));
            }
        }
        return list.get(0);
    }
}